source/graphics/graphics_window.cpp

```diff
@@ -6306,9 +6306,13 @@ These rotations are used in the MAYA viewing model.
 		norm_view[1] = view[1];
 		norm_view[2] = view[2];
 		/* establish right vector at view x up and normalise it and view */
-		if ((0.0 != normalize3(norm_view)) &&
-			cross_product3(norm_view, up, norm_right) &&
-			(0.0 != normalize3(norm_right)))
+		bool validView = 0.0 != normalize3(norm_view);
```

Comment:
Is this comparison to an actual `0.0` value what you want to do for a calculated value based on real numbers?

Reply:
I haven't changed the logic here; it uses a zinc internal function cross_product3 which no longer returns a value so can't be part of the condition.
Here, normalize3 returns the former magnitude of the vector and divides components by it only if non-zero.
Is there a compiler warning?

Reply:
Yes, compilers don't think it is a good idea to match `0.0` with a calculated value.  Neither do I.

Reply:
I've fixed this in the next PR See: https://github.com/cmiss/cmgui/pull/23/commits/d4a6d8692605140693a5bdca5b6fdf47ba402a66.
Note I've also caught up to develop.